applicationLibrary/molecularDynamics/multiSiteTests/LJMultisiteFunctorTest.cpp

```diff
@@ -0,0 +1,998 @@
+
+    for (size_t siteA = 0; siteA < numberOfSitesA; ++siteA) {
+      const auto exactSitePositionA = add(rotatedSitePositionsA[siteA], molA.getR());
+      for (size_t siteB = 0; siteB < numberOfSitesB; ++siteB) {
+        const auto exactSitePositionB = add(rotatedSitePositionsB[siteB], molB.getR());
+
+        const auto displacement = sub(exactSitePositionA, exactSitePositionB);
+        const auto distanceSquared = dot(displacement, displacement);
+
+        const auto sigmaSquared = PPL.getMixingData(siteTypesA[siteA], siteTypesB[siteB]).sigmaSquared;
+        const auto epsilon24 = PPL.getMixingData(siteTypesA[siteA], siteTypesB[siteB]).epsilon24;
+
+        const auto invDistSquared = 1. / distanceSquared;
+        const auto lj2 = sigmaSquared * invDistSquared;
+        const auto lj6 = lj2 * lj2 * lj2;
+        const auto lj12 = lj6 * lj6;
+        const auto lj12m6 = lj12 - lj6;  // = LJ potential / (4x epsilon)
+        const auto scalarMultiple = epsilon24 * (lj12 + lj12m6) * invDistSquared;
+        const auto force = autopas::utils::ArrayMath::mulScalar(displacement, scalarMultiple);
+
+        expectedForceA = add(expectedForceA, force);
+        if constexpr (newton3) {
+          expectedForceB = sub(expectedForceB, force);
+        }
+
+        const auto torqueOnA = cross(rotatedSitePositionsA[siteA], force);
+        expectedTorqueA = add(expectedTorqueA, torqueOnA);
+        if constexpr (newton3) {
+          const auto torqueOnB = cross(rotatedSitePositionsB[siteB], force);
+          expectedTorqueB = sub(expectedTorqueB, torqueOnB);
+        }
+
+        if constexpr (calculateGlobals) {
+          const auto shift6 = applyShift ? PPL.getMixingData(siteTypesA[siteA], siteTypesB[siteB]).shift6 : 0;
+          const auto shift = shift6 / 6.;
+          const auto epsilon4 = epsilon24 / 6.;
+
+          // only add half the potential energy if no newton3 is used (because we are missing half the interaction
+          // between the two molecules).
+          const auto potentialEnergy = newton3 ? epsilon4 * lj12m6 + shift : 0.5 * (epsilon4 * lj12m6 + shift);
+          const auto virialDimensionwiseContributions =
+              newton3 ? autopas::utils::ArrayMath::mul(displacement, force)
+                      : autopas::utils::ArrayMath::mulScalar(autopas::utils::ArrayMath::mul(displacement, force), 0.5);
+
+          expectedPotentialEnergySum += potentialEnergy;
+          expectedVirialSum += virialDimensionwiseContributions[0] + virialDimensionwiseContributions[1] +
+                               virialDimensionwiseContributions[2];
+        }
+      }
+    }
+  } else {
+    expectedForceA = {0, 0, 0};
+    expectedTorqueA = {0, 0, 0};
+    expectedForceB = {0, 0, 0};
+    expectedTorqueB = {0, 0, 0};
+  }
+
+  // calculate forces and torques using AoS functor
+
+  // create functor
+  mdLib::LJMultisiteFunctor<mdLib::MultisiteMoleculeLJ, applyShift, true, autopas::FunctorN3Modes::Both,
```

Comment:
Yes. No-Mixing was inherited from the single-site functors - and whilst it is possible to use this functor without mixing, the implementation is never used and doesn't really make sense (I'm not aware of simulations where pure LJ Multi-site molecules where every site is exactly the same - i.e. as I have implemented - unlike for single-site LJ molecules where no-mixing makes more sense). There are definitely optimisations to the functor which could be made which would be similar to no-mixing in single-site LJ molecules, but this is beyond the scope of this PR.

As such, I don't want to spend time creating tests for this case. If you are not happy this having this feature without tests, there are two alternatives to me creating tests:

1. Some warning in the constructor of the no-mixing version of the functor that says that the functor is not supported or tested and to use at their own risk.
2. Removal of the feature altogether.

Reply:
Just so I understand correctly. In the multisite setting, mixing refers not only to different sort of molecules but already different kind of sites on the same molecule. And as such mixing essentially needs to always be active, because the different atoms on say e.g. a $CH_4$ molecule have different sigmas, epsilons and so on?

I think, at least a warning that this feature is untested/experimental might still be a good idea.

Reply:
Yes you're correct. I do think there could be some benefit to some mode which contains only one type of molecule, even if the sites are mixed. But there are other issues with multi-site which should be addressed first.

I will add a warning